BuildingBlocks/src/BuildingBlocks.Infrastructure/EventBus/RabbitMQ/EventBusRabbitMQ.cs

```diff
@@ -210,20 +209,76 @@ private async Task ProcessEvent(string eventName, string message)
                 var concreteType = typeof(IIntegrationEventHandler<>).MakeGenericType(eventType);
 
                 var policy = EventBusRetryPolicyFactory.Create(
-                    _handlerRetryBehavior,
-                    (ex, _) => _logger.LogWarning(
-                        "The following error was thrown while executing '{eventHandlerType}':\n'{errorMessage}'\n{stacktrace}.\nAttempting to retry...",
-                        eventType.Name,
-                        ex.Message,
-                        ex.StackTrace)
-                    );
+                    _handlerRetryBehavior, (ex, _) => _logger.ErrorWhileExecutingEventHandlerType(eventName, ex.Message, ex.StackTrace!, ex));
 
                 await policy.ExecuteAsync(() => (Task)concreteType.GetMethod("Handle")!.Invoke(handler, new[] { integrationEvent })!);
             }
         }
         else
         {
-            _logger.LogWarning("No subscription for RabbitMQ event: '{EventName}'", eventName);
+            _logger.NoSubscriptionForEvent(eventName);
         }
     }
 }
+
+file static class LoggerExtensions
+{
+    private static readonly Action<ILogger, string, Exception> SOCKET_EXCEPTION =
+        LoggerMessage.Define<string>(
+            LogLevel.Warning,
+            new EventId(411326, "EventBusRabbitMQ.SocketException"),
+            "{exceptionString}"
+        );
+
+    private static readonly Action<ILogger, string, Exception> PUBLISHED_INTEGERATION_EVENT =
+        LoggerMessage.Define<string>(
+            LogLevel.Debug,
+            new EventId(585231, "EventBusRabbitMQ.PublishedIntegrationEvent"),
+            "Successfully published event with id '{integrationEventId}'."
+        );
+
+    private static readonly Action<ILogger, string, Exception> ERROR_WHILE_PROCESSING_INTEGRATION_EVENT =
+        LoggerMessage.Define<string>(
+            LogLevel.Error,
+            new EventId(702822, "EventBusRabbitMQ.ErrorWhileProcessingIntegrationEvent"),
+            "An error occurred while processing the integration event of type '{eventName}'."
+        );
+
+    private static readonly Action<ILogger, string, Exception> NO_SUBSCRIPTION_FOR_EVENT =
+        LoggerMessage.Define<string>(
+            LogLevel.Warning,
+            new EventId(980768, "EventBusRabbitMQ.NoSubscriptionForEvent"),
+            "No subscription for event: '{eventName}'."
+        );
+
+    private static readonly Action<ILogger, string, string, string, Exception> ERROR_WHILE_EXECUTING_EVENT_HANDLER_TYPE =
+        LoggerMessage.Define<string, string, string>(
+            LogLevel.Warning,
+            new EventId(288394, "EventBusRabbitMQ.ErrorWhileExecutingEventHandlerType"),
+            "The following error was thrown while executing '{eventHandlerType}':\n'{errorMessage}'\n{stackTrace}.\nAttempting to retry..."
+        );
+
+    public static void SocketException(this ILogger logger, string exceptionString, Exception e)
```

Comment:
No need to pass the `exceptionString`. You can as well get the Message property in here. 

Always try to make the use of these methods as easy as possible.